plugins/org.locationtech.udig.tool.edit/src/org/locationtech/udig/tools/edit/commands/SelectFeaturesAtPointCommand.java

```diff
@@ -193,6 +219,96 @@ private void runSelectionStrategies( IProgressMonitor monitor, FeatureIterator<S
                   false);
             }
         }
+        */
+        //code that enables selection of appropriate feature when multiple exist
+        //in the point where mouse click takes place
+        //******************************************************************/
+        List<SimpleFeature> featureList =new ArrayList<SimpleFeature>();
+        while (reader.hasNext()){
+            featureList.add(reader.next());
+        }      
+        final SimpleFeature[] features = featureList.toArray(new SimpleFeature[]{});
+
+        if (features.length == 1) {
+            for( SelectionStrategy selectionStrategy : strategies ) {
+                selectionStrategy.run(monitor, compositeCommand, parameters, features[0],
+                        true);
+            }
+        } else if (features.length > 1) {
+            PlatformGIS.syncInDisplayThread(new Runnable() {
+                @Override
+                public void run() {
+                	final String attribName = FeatureUtils.getActualPropertyName(
```

Comment:
Looks like it would take the first attribute that matches, in case of FeatureType `DataUtilities.createType("testType", "geometry:Polygon,name:String,timestamp:java.util.Date,NAME:String");`

it would return values of the first attribute **name** (because its in the definition before **NAME**), wouldn't it?